packages/lib-runtime/src/utils/basic-fetch.ts

```diff
@@ -0,0 +1,21 @@
+import { CustomError, applyDefaults } from '@monorepo/common';
+import type { ResourceFetch } from '@monorepo/common';
+
+class FetchError extends CustomError {
```

Comment:
This isn't exported because it's a custom error object the `basicFetch` implementation works with.

The `basicFetch` function is a fallback used when `PluginLoaderOptions.fetchImpl` is not specified (all properties within the `PluginLoaderOptions` object are optional by design).

Host apps normally want to use a custom fetch impl. and that impl. can throw whatever errors (declared in host app code) necessary.

TL;DR this is tied to `basicFetch` and not a general purpose HTTP fetch error object.